samples/bluetooth/fast_pair/locator_tag/src/ui/platform_dk.c

```diff
@@ -138,15 +152,37 @@ static void btn_handle(uint32_t button_state, uint32_t has_changed)
 		app_ui_request_broadcast(APP_UI_REQUEST_FP_ADV_MODE_CHANGE);
 	}
 
-	if (has_changed & button_state & APP_RING_BTN) {
-		app_ui_request_broadcast(APP_UI_REQUEST_RINGING_STOP);
+	if (has_changed & button_state & APP_RING_AND_MOTION_BTN) {
```

Comment:
Consider following flow:
1. Increase click counter
2. Reschedule the timeout job after each click
3. If there is no another click, the job will execute after its timeout, in which we can decide if the click count was proper and send the proper UI state
4. Clear click counter

As both jobs (timeout and button handler) are using workq iirc then it should work.
Moreover it allows us to extend the implementation later I think.

Reply:
I don't get it. What is the point of rescheduling the work after 2nd click? Shouldn't we send the proper ui request immediately after the 2nd click?

Reply:
I guess I could reschedule the work with `K_NO_WAIT` after the 2nd click to let it check the click count, send proper UI request and clear the counter. However, there's a race condition. The button click work may be submitted while handling the button work and we may get 3rd button press work before the work sending UI request is executed.